.github/workflows/main.yml

```diff
@@ -192,6 +192,58 @@ jobs:
           name: designcompose_m2repo
           path: build/designcompose_m2repo/
 
+  # These tests run on the local JVM and don't need the rust code, 
+  # so it makes sense to put them in a seperate job
+  roborazzi:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@8ca2b8b2ece13480cda6dacd3511b49857a23c09 # v2.5.1
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v3.5.2
+
+      - name: Set up Java
+        uses: actions/setup-java@cd89f46ac9d01407894225f350157564c9c7cee2 # v3.12.0
+        with:
+          distribution: "temurin"
+          java-version: "17"
+
+      - name: Validate Roborazzi
+        uses: gradle/gradle-build-action@ef76a971e2fa3f867b617efd72f2fbd72cf6f8bc # v2.8.0
+        with:
+          # Upload in the dependency-review workflow
+          dependency-graph: generate
+          arguments: verifyRoborazziDebug
+
+      - uses: actions/upload-artifact@v3
+        if: ${{ always() }}
+        with:
+          name: screenshot-diff
+          path: |
+            **/src/testDebug/roborazzi
+          retention-days: 30
+
+      - uses: actions/upload-artifact@v3
+        if: ${{ always() }}
+        with:
+          name: screenshot-diff-reports
+          path: |
+            **/build/reports
```

Comment:
I'd actually grabbed this job from your [Readme](https://github.com/takahirom/roborazzi#add-a-job-to-verify-screenshots). Maybe the same change should go there? And perhaps the job should also run `compareRoborazziDebug` to get all of the differences if verify fails? So that you can see them all rather than going one by one?

Reply:
Thanks! I agree, the same change should be made. As for `compareRoborazziDebug`, it really depends on how the test is written. If there's only one capture per test, it should suffice, as we can examine all changes. 

Just a thought, but perhaps we could create a JUnit rule using `CaptureResultReporter` that makes the test fail after capturing the images, without altering the core feature.

```java
class VerifyAfterTestRoborazziReporterRule : CaptureResultReporter {
    private val jsonOutputRoborazziCompareReporter = JsonOutputRoborazziReporter()
    private var captureResults = mutableListOf<CaptureResult>()
 
    override fun report(reportResult: CaptureResult) {
        jsonOutputRoborazziCompareReporter.report(reportResult)
        captureResults.add(reportResult)
    }
      
    fun onTestFinished() {
        if (captureResults.contains(Changed)) {
            throw AssertionError("xxxx")
        }
    }
}
```

Reply:
The whole code would be like this

```java
class ReporterRoborazziRule : TestWatcher() {
  class VerifyAfterTestRoborazziReporter : RoborazziOptions.CaptureResultReporter {
    private val jsonOutputRoborazziCompareReporter =
      RoborazziOptions.CaptureResultReporter.JsonOutputRoborazziReporter()
    private val captureResults = mutableListOf<Pair<CaptureResult, AssertionError>>()

    override fun report(reportResult: CaptureResult) {
      jsonOutputRoborazziCompareReporter.report(reportResult)
      captureResults.add(Pair(reportResult, AssertionError("xxxxx")))
    }

    fun onTestFinished() {
      val captureResultAssertionErrorPair =
        captureResults.firstOrNull { it.first is CaptureResult.Added || it.first is CaptureResult.Changed }
      if (captureResultAssertionErrorPair != null) {
        throw captureResultAssertionErrorPair.second
      }
      captureResults.clear()
    }
  }

  private var reporter: VerifyAfterTestRoborazziReporter? = null

  override fun starting(description: Description) {
    super.starting(description)
    reporter = VerifyAfterTestRoborazziReporter()
  }

  override fun finished(description: Description) {
    super.finished(description)
    reporter?.onTestFinished()
    reporter = null
  }

  override fun apply(base: Statement, description: Description): Statement {
    return RuleChain
      .outerRule(
        RoborazziRule(
          options = RoborazziRule.Options(
            roborazziOptions = RoborazziOptions(
              reportOptions = RoborazziOptions.ReportOptions(
                captureResultReporter = requireNotNull(reporter)
              )
            )
          )
        )
      )
      .apply(base, description)
  }
}
```

As for Roborazzi, we can't provide this as a library because users may want to customize RoborazziRule.Options themselves.

Reply:
We may need to consider alternative options due to the complexity involved 🤔

Reply:
Well how does the VerifyandRecord task work? I assumed that that would work the same way as running `Verify` and then `Record`, meaning it would fail and record fresh images for all of the tests. But am I wrong there? 

BTW, I'm fiddling with the screenshots in comments workflows in my [experimental fork](https://github.com/timothyfroehlich/automotive-design-compose/pull/98). I'll probably leave the workflows in this PR as is, since they'll do the basic verify test, and have a followup PR that adds the comments workflows and finishes up the verify job after. I'm keeping this one simple to allow my coworkers to comment on the basic implementation first.

Reply:
VerifyAndRecord works as follows:

```
test() {
  captureRoboImage("1.png") <- change detected
  captureRoboImage("2.png")
}
```

In test(), when captureRoboImage("1.png") detects a change, it will add 1_compare.png and update 1.png. Meanwhile, captureRoboImage("2.png") will be skipped.

I'm considering modifying this behavior in this PR, although it will make the implementation a bit more complex.
https://github.com/takahirom/roborazzi/pull/170/files

> followup PR

OK, I'll keep an eye on it. 👍

Reply:
Followup with the comment diffs is here: https://github.com/google/automotive-design-compose/pull/389. I made some relatively minor changes, mostly removing some of the artifact uploads, like the json files, that I don't currently have a use for. I also modified it to allow the diffs to work against the PR's base branch, instead of just main, to support feature branches and stacked PRs. I'll probably submit a PR to the repo with the same change for the base branch.

I think you should get those set up as generic actions too instead of just workflow examples. I think they're straightforward enough that they won't need to be customized too much. I might add a PR for that too.